pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PqlQueryResource.java

```diff
@@ -108,32 +109,39 @@ public String getQueryResponse(String pqlQuery, String traceEnabled, HttpHeaders
     }
     String tableName = TableNameBuilder.extractRawTableName(brokerRequest.getQuerySource().getTableName());
 
-    // Validate data access
-    AccessControl accessControl = _accessControlFactory.create();
-    if (!accessControl.hasDataAccess(httpHeaders, tableName)) {
-      return QueryException.ACCESS_DENIED_ERROR.toString();
-    }
+    String brokerLBurl = _pinotHelixResourceManager.getBrokerLoadBalancerAddress();
```

Comment:
We should get the list of broker instances for the table from pinot helix resource manager, and then use https on them if configured to do so. So, perhaps that calls for another config to use https when querying brokers.

Reply:
Alternative is for brokers to set their own configs in Helix instance config. We have gone that route with server http ports for example.